source/internationalDateOO.d

```diff
@@ -0,0 +1,123 @@
+
+import std.datetime : Clock, SysTime, UTC;
+
+immutable Language[string] translator;
+static this()
+{
+    import std.exception : assumeUnique;
+    auto result = 
+    [
+        "en" : new English(),
+        "de" : new German(),
+        "ru" : new Russian()
+    ];
+    translator = assumeUnique(result);
+}
+
+
+
+abstract class Language
+{
+    @property string pattern() const;
+    @property string now()     const;
+    
+    string minutes(long) const;
+    string hours(long) const;
+    
+    void writeDate(R)(ref R dst, SysTime time) const
+    {
+        dst.writeDate(time, Clock.currTime(UTC()));
+    }
+    
+    void writeDate(R)(ref R dst, SysTime time, SysTime today) const
+    {
+        import core.time : dur;
+        import std.format : formattedWrite;
+
+        auto tm = today - time;
+
+        if      (tm < dur!"seconds"(2)) dst.put(now);
+        else if (tm < dur!"hours"(1))   dst.formattedWrite(pattern, tm.total!"minutes", minutes(tm.total!"minutes"));
+        else                            dst.formattedWrite(pattern, tm.total!"hours",   hours  (tm.total!"hours"));
+    }
+}
+
+string toFuzzyDate(const Language lang, SysTime time)
+{
+    return lang.toFuzzyDate(time, Clock.currTime(UTC()));
+}
+
+string toFuzzyDate(const Language lang, SysTime time, SysTime now)
+{
+    import std.array : appender;
+    auto app = appender!string();
+    lang.writeDate(app, time, now);
+    return app.data;
+}
+

```

Comment:
That's what I actually thought. I kept only one file for demonstration.
The OO approach is easily extendable and as you can override the `writeDate` method, it can even react on very special languages, i.e. value dependent patterns.